microservices/nris_api/openshift/_python36_oracle.dc.json

```diff
@@ -0,0 +1,444 @@
+      "required": true
+    },
+    {
+      "name": "SUFFIX",
+      "displayName": "Name Suffix",
+      "description": "A suffix appended to all objects",
+      "required": true
+    },
+    {
+      "name": "DB_CONFIG_NAME",
+      "displayName": "Database Config Name",
+      "description": "Configuration name for the database",
+      "required": true
+    },
+    {
+      "name": "VERSION",
+      "required": true
+    },
+    {
+      "name": "JWT_OIDC_WELL_KNOWN_CONFIG",
+      "displayName": "Keycloak config URL",
+      "required": true
+    },
+    {
+      "name": "JWT_OIDC_AUDIENCE",
+      "displayName": "Keycloak clientId",
+      "required": true
+    },
+    {
+      "name": "BASE_PATH",
+      "required": false,
+      "value": "/"
+    },
+    {
+      "name": "MDS_NRIS_ODB_CREDENTIALS",
+      "required": false,
+      "value": "template.mds-nris-odb-credentials"
+    },
+    {
+      "name": "CPU_LIMIT",
+      "value": "200m"
+    },
+    {
+      "name": "MEMORY_LIMIT",
+      "value": "1Gi"
+    },
+    {
+      "name": "CPU_REQUEST",
+      "value": "150m"
+    },
+    {
+      "name": "MEMORY_REQUEST",
+      "value": "768Mi"
+    },
+    {
+      "name": "REPLICA_MIN",
+      "value": "1"
+    },
+    {
+      "name": "REPLICA_MAX",
+      "value": "1"
+    },
+    {
+      "name": "MDS_ELASTIC_SECRET",
+      "required": false,
+      "value": "template.mds-elastic-secret"
+    },
+    {
+      "name": "ELASTIC_ENABLED",
+      "value": "0"
+    },
+    {
+      "name": "ELASTIC_SERVICE_NAME",
+      "value": "MDS Dev"
+    },
+    {
+      "name": "REDIS_CONFIG_NAME",
+      "required": false,
+      "value": "template.mds-redis-secret"
+    },
+    {
+      "name": "ENVIRONMENT_NAME",
+      "displayName": "Environment Name (Environment Id)",
+      "description": "The name or Id of the environment.  This variable is used by the webhook integration to identify the environment in which the backup notifications originate.",
+      "required": false,
+      "value": "dev"
+    }
+  ],
+  "objects": [
+    {
+      "kind": "ImageStream",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "${NAME}",
+        "creationTimestamp": null,
+        "labels": {
+          "shared": "true"
+        },
+        "annotations": {
+          "description": "Keeps track of changes in the application image"
+        }
+      },
+      "spec": {
+        "lookupPolicy": {
+          "local": false
+        },
+        "tags": []
+      }
+    },
+    {
+      "kind": "DeploymentConfig",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "${NAME}${SUFFIX}",
+        "creationTimestamp": null,
+        "labels": null,
+        "annotations": {
+          "description": "Defines how to deploy the application server"
+        }
+      },
+      "spec": {
+        "strategy": {
+          "type": "Rolling",
+          "rollingParams": {
+            "post": {
+              "failurePolicy": "ignore",
```

Comment:
Don't you want to abort the deployment if the migration fails? If it silently ignores it we'll never know.

Also for Pre/Post hook. You may know better how alembic functions with the code but just wanted to point it out.

Coz with pre the behaviour is ->
The old app runs while the new migrations gets executed, if it works then new app gets scaled and old apps gets killed. Im not sure if alembic will complain about running migrations on a database while something is connected to it 🤷‍♂ 

and with post ->
The old app gets killed, the new one scales and then the migration is run. Now if the migration fails, you're still running new version of code and that may not work as expected.

Reply:
yea pre sounds better, i suppose that makes sense 'pre' means, before it's used, not before it's deployed. and yes, ignore was because i wanted to test other parts of the dc as well. it's now retry.